src/context/mod.rs

```diff
@@ -542,6 +542,22 @@ impl Context {
         action()
     }
 
+    /// Execute an arbitrary closure with the OpenGL context active and mutable access to
+    /// glium's internal state. Useful if another component needs to directly manipulate
+    /// OpenGL state or if another component already has.
+    ///
+    /// **`action` takes full responsibility for keeping OpenGL state in sync with the
+    /// context's state.**
+    #[inline]
+    pub unsafe fn exec_with_context<'a, T, F>(&self, action: F) -> T
+    where
+        T: Send + 'static,
+        F: FnOnce(&CommandContext<'_>) -> T + 'a,
```

Comment:
I opted not to make this a &mut as I couldn't see an immediate need for it, but in theory it wouldn't make it that much worse.

Reply:
Actually making a more general example for the bug report was a use for &mut.